### PR TITLE
ARC-107 update app

### PR DESCRIPTION
**Pull request by rachellerathbone**
All changes on this branch have already been approved (features branches that were approved and merged into `react-rewrite`. Feel free to pull the branch down and run over the usual testing to make sure everything looks ok (I did already and everything appears to be working).

### CODE REVIEW

File: src/backend/jira/client/jira-client-error.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/*
+ * An error wrapper that provides a more specific message for failed requests to the Jira API.
```
Comment (mboudreau) [inline src/backend/jira/client/jira-client-error.ts:2]: fairly certain this file was completely deleted.

File: src/backend/transforms/interfaces.ts [hunk 2]
```diff
@@ -9,7 +19,7 @@ export interface JiraPullRequest {
 	}
 }
 
-export interface JiraBuild {
+interface JiraBuild {
```
Comment (mboudreau) [inline src/backend/transforms/interfaces.ts:22]: Should probably still make this discoverable even if they're not being used currently.

**Review by mboudreau — CHANGES_REQUESTED** [review]
small changes.

File: src/backend/api/__snapshots__/api.test.ts.snap [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    },
+  ],
+  "viewerPermissionQuery": "{
+  viewer {
+    login
+    organization(login: \\"fusion-arc\\") {
+      viewerCanAdminister
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (rachellerathbone) [reply src/backend/api/__snapshots__/api.test.ts.snap:37]: Weird. I didn't manually update the snapshots. Just re-ran all the tests after moving files around. I'll take a look at this.

File: src/backend/jira/client/jira-client-error.ts (hunk 1 above)
Comment (rachellerathbone) [reply src/backend/jira/client/jira-client-error.ts:2]: Just checked out main and looks like it was. I'll delete it.

File: src/backend/api/__snapshots__/api.test.ts.snap (hunk 3 above)
[the remark this answers is not in the document]
Comment (rachellerathbone) [reply src/backend/api/__snapshots__/api.test.ts.snap:37]: You're definitely right about there being missing snapshots. This is what is should be:
```
// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API Authentication should return 200 if a valid token is provided 1`] = `Object {}`;

exports[`API Authentication should return 200 if token belongs to an admin 1`] = `Object {}`;

exports[`API Authentication should return 401 if the GraphQL query returns errors 1`] = `
Object {
  "errors": Array [
    Object {
      "extensions": Object {
        "code": "undefinedField",
        "fieldName": "foo",
        "typeName": "User",
      },
      "locations": Array [
        Object {
          "column": 5,
          "line": 4,
        },
      ],
      "message": "Field 'foo' doesn't exist on type 'User'",
      "path": Array [
        "query",
        "viewer",
        "foo",
      ],
    },
  ],
  "viewerPermissionQuery": "{
  viewer {
    login
    organization(login: \\"fusion-arc\\") {
      viewerCanAdminister
    }
  }
}
",
}
`;

exports[`API Authentication should return 401 if the returned organization is null 1`] = `
Object {
  "error": "Unauthorized",
  "message": "Token provided does not have required access",
}
`;

exports[`API Authentication should return 401 if the token is invalid 1`] = `
Object {
  "documentation_url": "https://developer.github.com/v4",
  "message": "Bad credentials",
}
`;

exports[`API Authentication should return 404 if no token is provided 1`] = `Object {}`;

exports[`API Endpoints installation should return 404 if no installation is found 1`] = `Object {}`;

exports[`API Endpoints installation should return information for an existing installation 1`] = `
Object {
  "connections": Array [
    Object {
      "isGlobalInstall": false,
      "updated_at": "Invalid Date",
    },
  ],
  "failedConnections": Array [],
  "hasConnections": true,
  "host": "https://test-atlassian-instance.net",
  "installationId": "1234",
  "repoSyncState": "http://127.0.0.1/api/1234/repoSyncState.json",
}
`;

exports[`API Endpoints repoSyncState should return 404 if no installation is found 1`] = `Object {}`;

exports[`API Endpoints repoSyncState should return the repoSyncState information for an existing installation 1`] = `""`;

exports[`API Endpoints sync should reset repoSyncState if asked to 1`] = `"{\\"message\\":\\"Successfully (re)started sync for 1234\\"}"`;

exports[`API Endpoints sync should return 404 if no installation is found 1`] = `"Not Found"`;

exports[`API Endpoints sync should trigger the sync or start function 1`] = `"{\\"message\\":\\"Successfully (re)started sync for 1234\\"}"`;

exports[`API Endpoints sync should trigger the sync or start function 2`] = `"{\\"message\\":\\"Successfully (re)started sync for 1234\\"}"`;

exports[`API Endpoints undo and complete - nonprod should not migrate an installation 1`] = `"Error trying to complete migration: TypeError: Cannot read property 'devinfo' of undefined"`;

exports[`API Endpoints undo and complete - nonprod should not migrate an installation 2`] = `[MockFunction]`;

exports[`API Endpoints undo and complete - nonprod should not undo a migration 1`] = `"Error trying to undo migration: TypeError: Cannot read property 'devinfo' of undefined"`;

exports[`API Endpoints undo and complete - nonprod should not undo a migration 2`] = `[MockFunction]`;

exports[`API Endpoints undo and complete - prod should return 404 if no installation is found 1`] = `"Not Found"`;

exports[`API Endpoints verify should return 'Installation already enabled' 1`] = `"Installation already enabled"`;
```

However, even the original version is checking for a 401.

[the remark this answers is not in the document]
Comment (rachellerathbone) [reply src/backend/api/__snapshots__/api.test.ts.snap:37]: I reverted the snapshot back. As for the issue with the 401, if there's something wrong with the associated test we should create an issue to fix it (this PR is already too big 😬 )

**Comment by rachellerathbone** [comment]
Closing because there are now too many conflicts. Created https://softwareteams.atlassian.net/jira/software/projects/ARC/boards/385/backlog?selectedIssue=ARC-672 to revisit at a later date.

**Pull request closed without merging**